streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala

```diff
@@ -84,7 +86,10 @@ trait StreamingListener {
  * @param numBatchInfos Number of last batches to consider for generating statistics (default: 10)
  */
 @DeveloperApi
-class StatsReportListener(numBatchInfos: Int = 10) extends StreamingListener {
+class StatsReportListener(numBatchInfos: Int) extends StreamingListener {
+
+  def this() = this(10)
```

Comment:
The previous one could not be instantiated by reflection due to having no
zero-argument constructor. This provides a built-in class that can be
created at initialization without changing the functionality of the class.
On May 26, 2015 2:23 AM, "Tathagata Das" <notifications@github.com> wrote:

> In
> streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala
> <https://github.com/apache/spark/pull/6380#discussion_r31012977>:
>
> > @@ -84,7 +86,10 @@ trait StreamingListener {
> >   * @param numBatchInfos Number of last batches to consider for generating statistics (default: 10)
> >   */
> >  @DeveloperApi
> > -class StatsReportListener(numBatchInfos: Int = 10) extends StreamingListener {
> > +class StatsReportListener(numBatchInfos: Int) extends StreamingListener {
> > +
> > +  def this() = this(10)
>
> How is this different from previous one?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/6380/files#r31012977>.
>